### PR TITLE
Open API Update - Sync with Open API PR 249

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
## Description

- Sync with Open API Spec update https://github.com/okta/openapi/pull/249.
- Resolves [OKTA-308695](https://oktainc.atlassian.net/browse/OKTA-308695).
- Resolves [OKTA-308697](https://oktainc.atlassian.net/browse/OKTA-308697).

**TODO:** Cut tag `1.2.4` out of `okta-commons`.

## Category

- [ ] Bugfix
- [ ] Enhancement
- [ ] New Feature
- [ ] Configuration Change
- [ ] Versioning Change
- [ ] Unit Test(s)
- [ ] Documentation

### CODE REVIEW

**Comment by bdemers** [comment]
Looks good, but we should probably add a test to make sure the dates get formatted correctly: https://github.com/okta/okta-sdk-java/blob/dev_openapi_v2.0.0_major_rel/impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy#L44

**Comment by arvindkrishnakumar-okta** [comment]
> Looks good, but we should probably add a test to make sure the dates get formatted correctly: https://github.com/okta/okta-sdk-java/blob/dev_openapi_v2.0.0_major_rel/impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy#L44

Added UTs.

File: impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy [hunk 1]
```diff
@@ -145,4 +151,38 @@ class LogsTest {
...
     }
+
+    @Test
+    void testGetLogsBetweenDates() {
+
+        // get the events between the 'until' and 'since' date params
+        Date untilDate = Date.from(Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse("2017-11-30T21:16:00.081Z")))
+        Date sinceDate = Date.from(Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse("2017-11-30T21:15:16.838Z")))
+
+        List<LogEvent> logs = client.getLogs(untilDate, sinceDate, null, null, null).stream().collect(Collectors.toList())
```
Comment (bdemers) [inline impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy:162]: This method needs to verify the format of `untilDate` and/or `sinceDate` that would be sent over the wire, either with `MockClient.withMockResponse(...)` or `verify(mockClient.mockRequestExecutor).*`
(or something similar)

Comment (arvindkrishnakumar-okta) [reply impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy]: Updated.

File: impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy [hunk 2]
```diff
@@ -145,4 +153,50 @@ class LogsTest {
...
+        Date sinceDate = Date.from(Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse("2017-11-30T21:15:16.838Z")))
+        Date untilDate = Date.from(Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse("2017-11-30T21:16:00.081Z")))
+        
+        // get log events between dates
+        List<LogEvent> logs = client.getLogs(sinceDate, untilDate, null, null, null).stream().collect(Collectors.toList())
+        logs.forEach { assertThat it, instanceOf(LogEvent) }
+
+        ArgumentCaptor<Request> argument = ArgumentCaptor.forClass(Request.class)
+        verify(client.mockRequestExecutor).executeRequest(argument.capture())
+        assertThat argument.getValue().queryString.since, instanceOf(String)
```
Comment (bdemers) [inline impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy:169]: Check the specific format of this string to make sure it matches what the Okta API expects

Comment (arvindkrishnakumar-okta) [reply impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy:169]: I see that date is not handled correctly and will need to fix this issue (created https://oktainc.atlassian.net/browse/OKTA-308568).

File: pom.xml [hunk 3]
```diff
@@ -40,7 +40,7 @@
         <bouncycastle.version>1.64</bouncycastle.version>
         <jjwt.version>0.11.1</jjwt.version>
         <okta.sdk.previousVersion>1.5.4</okta.sdk.previousVersion>
-        <otka.commons.version>1.2.3</otka.commons.version>
+        <otka.commons.version>1.2.4</otka.commons.version>
```
Comment (arvindkrishnakumar-okta) [inline pom.xml:43]: **TODO**: Cut tag `1.2.4` out of `okta-commons`.

File: impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply impl/src/test/groovy/com/okta/sdk/impl/resource/log/LogsTest.groovy]: Fixed in PR - https://github.com/okta/okta-commons-java/pull/36.

File: impl/src/test/groovy/com/okta/sdk/impl/ds/DefaultCacheKeyTest.groovy [hunk 4]
```diff
@@ -51,6 +54,19 @@ class DefaultCacheKeyTest {
         assertEquals cacheKey.toString(), "https://mysite.com?key_one=value_one&key_three=value_three&key_two=value_two"
     }
 
+    @Test
+    void testWithDateInQueryString() {
+        def isoDateString = "2017-11-30T21:15:16.838Z"
+        Date date = Date.from(Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse(isoDateString)))
+
+        def qs = new QueryString([
+            "key_one":date
```
Comment (bdemers) [inline impl/src/test/groovy/com/okta/sdk/impl/ds/DefaultCacheKeyTest.groovy:63]: nit:
```suggestion
            "key_one": date
```

**Comment by arvindkrishnakumar-okta** [comment]
> one nit, but looks good!

Addressed this.

**Pull request merged**